### ISSUE TITLE
Can I ask if partial convolution can be applied to super-resolution tasks?

### ISSUE THREAD

**Issue by HaoLongfei-1** [ivanstepanovftw/masked_torch#1]
Since in my dataset, there are missing values, which I have replaced with zeros, is there a way to ignore the effect of these zeros for super resolution.

**Comment by ivanstepanovftw** [comment]
If you know which parts are missing, you can make relevant mask of same image shape that you need to feed along with inputs.

For example, in the input batch you have 1 RGB image of size 800x600, and you know that it has missing values at the bottom half, then you need to make a mask with zeros at missing places and pass both input and mask at the same time:
```python
input = torch.randn(1, 3, 600, 800)  # (Batch x Channels x Height x Width)
input_mask = torch.ones_like(input)
input_mask[:, :, 300:, :] = 0  # top left pixel is at (0, 0) position
output, output_mask = model(input * input_mask, input_mask)
```

If you do not know where exactly images are corrupted, you can use regular convolution just as fine.

And here is how you can define the model:
https://github.com/ivanstepanovftw/masked_torch/blob/c4d1052a418f12280b156bfbd67b11e6f0440e16/examples/visualize_activations_2d.py#L123-L133
